ui/components/ui/disclosure/disclosure.js

```diff
@@ -1,9 +1,53 @@
 import React, { useState, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
-import { Icon, IconName, IconSize } from '../../component-library';
+import { Icon, IconName, IconSize, Text } from '../../component-library';
+import { Color, TextVariant } from '../../../helpers/constants/design-system';
+import { DisclosureVariant } from './disclosure.constants';
 
-const Disclosure = ({ children, title, size }) => {
+/**
+ * @param {string} variant
+ * @param {string} title
+ * @param {string} size
+ * @returns {JSX.Element}
+ */
+const renderSummaryByType = (variant, title, size) => {
+  switch (variant) {
+    case DisclosureVariant.Arrow: {
+      const textVariant =
+        size === 'small' ? TextVariant.bodySm : TextVariant.bodyMd;
+
+      return (
+        <summary className="disclosure__summary is-arrow">
+          <Text color={Color.primaryDefault} variant={textVariant}>
+            {title}
+          </Text>
+          <Icon
+            className="disclosure__summary--icon"
+            color={Color.primaryDefault}
+            name={IconName.ArrowUp}
+            size={IconSize.Sm}
+            marginInlineStart={2}
+          />
+        </summary>
+      );
+    }
+    default:
+      return (
+        <summary className="disclosure__summary">
+          <Icon
+            className="disclosure__summary--icon"
+            name={IconName.Add}
+            size={IconSize.Sm}
+            marginInlineEnd={2}
+          />
+          {title}
+        </summary>
+      );
+  }
+};
+
+const Disclosure = ({ children, title, size, type }) => {
```

Comment:
I was also wondering if it made sense to change the `type` prop to `variant` then we have this pattern of `ComponentName` + `PropName` for the enum? Sorry if I was unclear on my last review

Reply:
whoops missed those. updated. thank you! [4e929c7](https://github.com/MetaMask/metamask-extension/pull/19770/commits/4e929c77ad026928625d1dd40940e4ce447b2dfe) cc: @georgewrmarshall